packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -102,4 +242,104 @@ protected function getDefinition(int $domainId, string $index): array
 
         return $definition;
     }
+
+    /**
+     * @param array $array
+     * @return bool
+     */
+    protected function recursiveArraySort(array &$array)
+    {
+        foreach ($array as &$value) {
+            if (is_array($value)) {
+                $this->recursiveArraySort($value);
+            }
+        }
+        return ksort($array);
+    }
+
+    /**
+     * @param array $currentIndex
+     * @return array
+     */
+    protected function prepareCurrentIndexDefinition(array $currentIndex): array
+    {
+        $currentIndexDefinition = reset($currentIndex);
+        $currentIndexDefinition['settings'] = $currentIndexDefinition['settings']['index'];
+        $currentIndexDefinition['settings']['index'] = [
+            'number_of_replicas' => (int)$currentIndexDefinition['settings']['number_of_replicas'],
```

Comment:
So this means elastic returns `string` and when string is provided, then it doesn't accept it? Strange behavior

Reply:
please come with a better name for this function as this is not a preparation.... it's preparation for comparing... or something like that

Reply:
I changed the name. The thing is that we have the information in *.json as integer and Elastic saves it to itself as a string. So to compare it strictly it necessary to retype it to int or change the way we have it in *.json